src/Workspaces/Core/Portable/Remote/DefaultRemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal partial class DefaultRemoteHostClientServiceFactory
+    {
+        public class RemoteHostClientService : IRemoteHostClientService
+        {
+            private readonly Task<RemoteHostClient> _instanceTask;
+
+            public RemoteHostClientService(Workspace workspace)
+            {
+                var remoteHostClientFactory = workspace.Services.GetService<IRemoteHostClientFactory>();
+                if (remoteHostClientFactory == null)
+                {
+                    // no implementation of remote host client
+                    return;
+                }
+
+                _instanceTask = Task.Run(() => remoteHostClientFactory.CreateAsync(workspace, CancellationToken.None), CancellationToken.None);
```

Comment:
You're creating two non-cancellable tasks here: "Task.Run(() => remoteHostClientFactory.CreateAsync(workspace, CancellationToken.None), CancellationToken.None)"

I don't see why these shouldn't be cancellable.  For example, you get passed cancellation tokens in " GetRemoteHostClientAsync(CancellationToken cancellationToken)".  So why not use that cancellation token so you can cancel this work if necessary?

Either this work is cancellable (which is indicated by it taking cancellation tokens) and we should actually pass along appropriate cancellation tokens.  Or it's non-cancellable, and it shoudl not have cancellation tokens in the signature.

One way you could do this would be to move to AsyncLazy here.